### PR TITLE
Switch to using stream-metadata-go library

### PR DESCRIPTION
**Pull request by cgwalters**
After we have a shared Go library for streams, we can use it
in e.g. coreos-assembler/mantle as well as openshift-install.

### CODE REVIEW

**Comment by cgwalters** [comment]
I moved most of the metadata definitions here to https://github.com/cgwalters/stream-metadata-go per discussion in https://github.com/coreos/fedora-coreos-tracker/issues/98#issuecomment-683150940

So if this PR and approach looks good I propose we move github.com/cgwalters/stream-metadata-go into coreos/ and then look to merge this!

**Comment by bgilbert** [comment]
I'm in favor.  If we're moving these structs out to their own library, though, I think we should probably clean up the struct naming.  It isn't very consistent with the terminology used elsewhere.

Also, maybe we should use a map of platforms rather than hardcoding the platform list into the struct?  As is, we'll need to revendor the library every time we add a platform.

**Comment by cgwalters** [comment]
>  I think we should probably clean up the struct naming. It isn't very consistent with the terminology used elsewhere.

Do you have a specific example?  Actually if you want feel free to just push code to master in https://github.com/cgwalters/stream-metadata-go - I just invited you as a collaborator.  (Or do PRs as you prefer)

**Comment by cgwalters** [comment]
> Also, maybe we should use a map of platforms rather than hardcoding the platform list into the struct? As is, we'll need to revendor the library every time we add a platform.

Some of them need special handling (mostly the public cloud ones like AWS/Azure/GCP) - but maybe the ones that are just "an image file" could be part of a string map I guess?

**Comment by bgilbert** [comment]
Okay, I've pushed a bunch of refactoring to that repo.

One open question: calling the package `stream` is a bit of a misnomer, since it also includes structs for release metadata.  Should we move `stream.go` and `release.go` into their own subpackages, and/or rename the package and repo e.g. to `coreos-metadata-go`?

(eek, [coreos-metadata](https://github.com/coreos/coreos-metadata).)

**Comment by cgwalters** [comment]
> One open question: calling the package stream is a bit of a misnomer, since it also includes structs for release metadata.

Yeah, separate packages seems cleaner I guess.

**Comment by cgwalters** [comment]
Pushed https://github.com/cgwalters/stream-metadata-go/pull/1

**Comment by cgwalters** [comment]
OK this should be close to ready, just have the two dependency PRs outstanding.

**Comment by cgwalters** [comment]
OK lifting draft!

File: main.go [hunk 1]
```diff
@@ -131,54 +182,45 @@ func releaseToStream(releaseArch *ReleaseArch, rel Release) StreamArch {
...
-		}
-		artifacts.Virtualbox = &virtualbox
-	}
+	// if releaseArch.Media.Virtualbox != nil {
+	// 	virtualbox := stream.PlatformArtifacts{
+	// 		Release: rel.Release,
+	// 		Formats: releaseArch.Media.Virtualbox.Artifacts,
+	// 	}
+	// 	artifacts.Virtualbox = &virtualbox
+	// }
```
Comment (cgwalters) [inline main.go:204]: Ah right I meant to hoist out the virtualbox disablement too.  Eh.  I can if you guys care.

Comment (bgilbert) [reply main.go:204]: A separate commit would be nice, if convenient.  A separate PR isn't necessary.

Comment (cgwalters) [reply main.go:204]: OK did that and also a separate commit for the url function.

File: main.go [hunk 2]
```diff
@@ -258,20 +300,20 @@ func run() error {
...
 	if err = decoder.Decode(&rel); err != nil {
 		return fmt.Errorf("Error while decoding json: %v", err)
 	}
 
-	streamArch := make(map[string]*StreamArch)
+	streamArch := make(map[string]stream.Arch)
 	for arch, releaseArch := range rel.Architectures {
-		archContent := releaseToStream(releaseArch, rel)
-		streamArch[arch] = &archContent
+		archContent := releaseToStream(&releaseArch, rel)
```
Comment (bgilbert) [inline main.go]: Intermediate `archContent` variable can be dropped.

**Review by bgilbert — APPROVED** [review]

**Pull request merged**